packages/api/models/rules.js

```diff
@@ -159,16 +159,25 @@ class Rule extends Manager {
           return (mapping.EventSourceArn === item.rule.value);
         });
       if (mappingExists) {
-        item.rule.arn = mappingExists.UUID;
-        return item;
+        const mappingEnabled = listData.EventSourceMappings
+          .find((mapping) => { // eslint-disable-line arrow-body-style
+            return (mapping.EventSourceArn === item.rule.value &&
+                    mapping.State === 'Enabled');
+          });
+
+        if (mappingEnabled) {
+          item.rule.arn = mappingEnabled.UUID;
+          return item;
+        }
+        await aws.lambda().deleteEventSourceMapping({ UUID: mappingExists.UUID }).promise();
```

Comment:
The mapping could be shared by other rules, use rules.deleteKinesisEventSource() method instead.

Reply:
It's in a new file. The stream is created at the beginning of integration tests, and destroyed at the end.  
https://github.com/nasa/cumulus/pull/455/files#diff-f4a144d837fc6c0abc86b61b63973879R10

Reply:
This api/models/rules.js is part of the api, not only for integration tests.